CMakeLists.txt

```diff
@@ -227,9 +227,44 @@ include(Compilers)
 message(STATUS "Configuring build settings...")
 include(BuildSettings)
 
+if (MSVC)
+    # TODO: These msvc specific settings probably belong in BuildSettings. 
+    # But I am not sure where to put them, so I will leave them in the obvious
+    # place for now.
+
+    # set debug postfix so debug library and executable artifact names do not
+    # conflict with release artifacts
+    set(CMAKE_DEBUG_POSTFIX "_d")
+
+    # VTune is much more helpful when debug information is included in the 
+    # generated release code. 
+    add_definitions(/Zi)
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /debug")
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /debug")
+    set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} /debug")
+    foreach(flag_var
+        CMAKE_CXX_FLAGS CMAKE_CXX_FLAGS_DEBUG CMAKE_CXX_FLAGS_RELEASE
+        CMAKE_CXX_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_RELWITHDEBINFO)
+        if(${flag_var} MATCHES "/RTC1")
+            string(REGEX REPLACE "/RTC1" "" ${flag_var} "${${flag_var}}")
```

Comment:
What is the problem with this flag?

Reply:
Sorry, that was a copy/paste of cmake logic I have carried around for a long time. I used a cmake 3 approach and set these /Zi and /debug as options on the timemory-compile-options target. And I removed the logic for removing /RTC1 as a flag (I am not sure what it does.)